iphone/Maps/Classes/CustomViews/MapViewControls/Search/TableView/MWMSearchCell.mm

```diff
@@ -45,9 +48,19 @@ - (void)config:(search::Result &)result
   size_t const rangesCount = result.GetHighlightRangesCount();
   for (size_t i = 0; i < rangesCount; ++i)
   {
-    pair<uint16_t, uint16_t> const range = result.GetHighlightRange(i);
-    [attributedTitle addAttributes:selectedTitleAttributes
-                             range:NSMakeRange(range.first, range.second)];
+    pair<uint16_t, uint16_t> const & range = result.GetHighlightRange(i);
+
+    if (range.first + range.second <= title.length)
+    {
+      [attributedTitle addAttributes:selectedTitleAttributes range:NSMakeRange(range.first, range.second)];
+    }
+    else
+    {
+      [Statistics logEvent:@"Incorrect_highlight_range" withParametrs:@{@"range.first" : @(range.first),
+                                                                        @"range.second" : @(range.second),
+                                                                        @"string" : title}];
```

Comment:
Тут без разницы, потому что все равно в NSString преобразуется по итогу.

> On Apr 5, 2016, at 21:31, Vlad Mihaylenko <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/CustomViews/MapViewControls/Search/TableView/MWMSearchCell.mm <https://github.com/mapsme/omim/pull/2763#discussion_r58591428>:
> 
> > @@ -45,9 +48,19 @@ - (void)config:(search::Result &)result
> >    size_t const rangesCount = result.GetHighlightRangesCount();
> >    for (size_t i = 0; i < rangesCount; ++i)
> >    {
> > -    pair<uint16_t, uint16_t> const range = result.GetHighlightRange(i);
> > -    [attributedTitle addAttributes:selectedTitleAttributes
> > -                             range:NSMakeRange(range.first, range.second)];
> > +    pair<uint16_t, uint16_t> const & range = result.GetHighlightRange(i);
> > +
> > +    if (range.first + range.second <= title.length)
> > +    {
> > +      [attributedTitle addAttributes:selectedTitleAttributes range:NSMakeRange(range.first, range.second)];
> > +    }
> > +    else
> > +    {
> > +      [Statistics logEvent:@"Incorrect_highlight_range" withParametrs:@{@"range.first" : @(range.first),
> > +                                                                        @"range.second" : @(range.second),
> > +                                                                        @"string" : title}];
> Тогда и тут нужно reault.GetString() наверное
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2763/files/d1428fbbf95465af9c816393e35d88e2bf56f9d0#r58591428>